CIP-0030/README.md

```diff
@@ -247,6 +245,30 @@ Errors: `APIError`, `TxSendError`
 
 As wallets should already have this ability, we allow dApps to request that a transaction be sent through it. If the wallet accepts the transaction and tries to send it, it shall return the transaction id for the dApp to track. The wallet is free to return the `TxSendError` with code `Refused` if they do not wish to send it, or `Failure` if there was an error in sending it (e.g. preliminary checks failed on signatures).
 
+
+
+## Events
+
+In addition to the API methods that are actively called, the connector also must emit the following events. All methods events are required to be implemented.
+
+TODO: event emission method? Possible methods:
```

Comment:
I think it's best to emit `wallet_disconnected` on the `wallet` object. I also think there has to be a method to unsubscribe.

Regarding the method/naming:
* To not confuse with DOM events, I would rule out `window.addEventListener`.
* To not confuse with cross-window/context communication, I would rule out `postMessage`

This leaves us with the following options:
* `wallet.on(eventName, handler): void` and `wallet.off(eventName, handler): void` - probably the most used pattern besides addEventListener/removeEventListener. I personally like this one best.
* `wallet.on(eventName, handler): Unsubscribe`, `type Unsubscribe = () => void`
* `wallet.onEvent(handler): Unsubscribe`

Also, I suggest documenting that wallets will automatically unsubscribe all event listeners when disconnected (so dApps know they don't have to do that).

Reply:
> wallet.on(eventName, handler): void and wallet.off(eventName, handler): void - probably the most used pattern besides addEventListener/removeEventListener. I personally like this one best.

:+1:  for this suggestion, `on/off` or even `addListener/removeListener` are indeed the most commonly seen practices and easy to remember.

Reply:
`wallet.on(eventName, handler): Unsubscribe, type Unsubscribe = () => void` has a nice advantage, though - doesn't need referencing `handler` to unsubscribe, which leads to an API that is easier to use and easier to follow (IMO) as there is no second method that widens initial API surface. Effectively, it is at least harder to unsubscribe if there was no subscription in the first place. Additionally - in case of testing functions depending on this API - it will be easier to fake it, e.g. by wrapping an observable.

And there are quite known APIs that utilize similar pattern:
  - React in its `useEffect` hook
  - rxjs in `Observable` constructor

Reply:
@kapke imo that doesn't make sense. The reason why React and Observable does it like that is because they housekeep the event, which makes sense if you are observing the values. In React, for example, you can only (unconditionally) call `useEffect` in a component/hook, which is why they need to do the clean up in that context. These limitations does not apply to the Cardano Bridge API, which is only limited to the scope and life-cycle of the window. The way that nami implements the events makes much more sense to me, by just re-using the standard `Event` object.